lang/java/reef-tang/tang/src/test/java/org/apache/reef/tang/implementation/avro/TestClassHierarchyRoundTrip.java

```diff
@@ -30,372 +30,520 @@
 import java.io.IOException;
 
 public class TestClassHierarchyRoundTrip extends TestClassHierarchy {
```

Comment:
It is better to add new tests for Avro so that to keep the tests for proto buffer. The reason is proto buffer and Avro may co-exists for a while. I am encountering some issues in .Net Avro library for supporting List and nullable.